internal/olm/client/client.go

```diff
@@ -48,6 +48,22 @@ var ErrOLMNotInstalled = errors.New("no existing installation found")
 
 var Scheme = scheme.Scheme
 
+// custom error struct to capture deployment errors
+// while verifying CSV installs.
+type deploymentError struct {
+	name  string
+	issue string
+}
+type deploymentVerifyError []deploymentError
+
+func (e deploymentVerifyError) Error() string {
+	var sb strings.Builder
+	for _, i := range e {
+		sb.WriteString(i.name + " has errors: \n" + i.issue + "\n")
+	}
+	return sb.String()
+}
```

Comment:
There are really 2 types of errors here, `deploymentError` and `podError`, that could be organized like:
```suggestion
// custom error type to capture resource errors
// while verifying CSV installs.
type resourceError struct {
	name  string
	issue string
}

type deploymentErrors []deploymentError


func (e deploymentErrors) Error() string {
	var sb strings.Builder
	for _, i := range e {
		sb.WriteString(fmt.Sprintf("deployment %s has error: %s\n", i.name, i.issue)
		for _, j := range i.podErrors {
			sb.WriteString(j.Error())
		}
	}
	return sb.String()
}

type deploymentError struct {
	resourceError
	podErrors podErrors
}

type podErrors []podError

func (e podErrors) Error() string {
	var sb strings.Builder
	for _, i := range e {
		sb.WriteString(fmt.Sprintf("\tpod %s has error: %s\n", i.name, i.issue)
	}
	return sb.String()
}

type podError struct {
	resourceError
}
```

that way we're not muddling error semantics.